### PR TITLE
Hieu/fix 46 sdk

### PR DESCRIPTION
**Pull request by lehieuhust**
Migrate to Cosmos v46

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -364,6 +364,8 @@ func NewQuicksilver(
 	scopedICAControllerKeeper := app.CapabilityKeeper.ScopeToModule(icacontrollertypes.SubModuleName)
 	scopedICAHostKeeper := app.CapabilityKeeper.ScopeToModule(icahosttypes.SubModuleName)
 	scopedInterchainStakingKeeper := app.CapabilityKeeper.ScopeToModule(interchainstakingtypes.ModuleName)
+	scopedIBCMockKeeper := app.CapabilityKeeper.ScopeToModule(ibcmock.ModuleName)
+	ScopedICAMockKeeper := app.CapabilityKeeper.ScopeToModule(icahosttypes.SubModuleName)
```
Comment (nghuyenthevinh2000) [inline app/app.go]: should be: "scopedICAMockKeeper"

File: app/app.go [hunk 2]
```diff
@@ -613,7 +615,7 @@ func NewQuicksilver(
 		authzmodule.NewAppModule(appCodec, app.AuthzKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry),
 		// ibc modules
 		ibc.NewAppModule(app.IBCKeeper),
-		transferModule,
+		// transferModule,
```
Comment (nghuyenthevinh2000) [inline app/app.go:618]: should have tried to solve the error

File: x/interchainstaking/ibc_module.go [hunk 3]
```diff
@@ -266,7 +266,7 @@ func (im IBCModule) OnRecvPacket(
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
 ) ibcexported.Acknowledgement {
-	return channeltypes.NewErrorAcknowledgement("cannot receive packet via interchain accounts authentication module")
+	return channeltypes.NewErrorAcknowledgement(types.Errors{})
```
Comment (nghuyenthevinh2000) [inline x/interchainstaking/ibc_module.go]: https://github.com/cosmos/ibc-go/blob/48a6ae512b4ea42c29fdf6c6f5363f50645591a2/modules/core/04-channel/types/acknowledgement_test.go#L47

Comment (nghuyenthevinh2000) [reply x/interchainstaking/ibc_module.go]: fmt.Errorf("cannot receive packet via interchain accounts authentication module")

File: x/interchainstaking/keeper/proposal_handler.go [hunk 4]
```diff
@@ -82,7 +82,7 @@ func HandleRegisterZoneProposal(ctx sdk.Context, k Keeper, p *types.RegisterZone
 }
 
 func (k Keeper) registerInterchainAccount(ctx sdk.Context, connectionId string, portOwner string) error {
-	if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, portOwner); err != nil {
+	if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, portOwner, ""); err != nil {
```
Comment (nghuyenthevinh2000) [inline x/interchainstaking/keeper/proposal_handler.go]: check mau trong github.com/cosmos/ibc-go

File: x/interchainstaking/types/codec.go [hunk 5]
```diff
@@ -39,8 +39,8 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 func init() {
 	cryptocodec.RegisterCrypto(amino)
 	govtypes.RegisterProposalType(ProposalTypeRegisterZone)
-	govtypes.RegisterProposalTypeCodec(&RegisterZoneProposal{}, "cosmos-sdk/RegisterZoneProposal")
+	amino.RegisterConcrete(&RegisterZoneProposal{}, "cosmos-sdk/RegisterZoneProposal", nil)
```
Comment (nghuyenthevinh2000) [inline x/interchainstaking/types/codec.go]: https://github.com/notional-labs/evmos/blob/dang/fix-sdk46-test/x/erc20/types/proposal.go#L33

Comment (nghuyenthevinh2000) [reply x/interchainstaking/types/codec.go]: try to run and will face amino is sealed

File: app/app.go (hunk 1 above)
Comment (lehieuhust) [reply app/app.go]: Fix in https://github.com/notional-labs/quicksilver/commit/aab37f851670fde663ed1b04b299e6be033aed8d

File: x/interchainstaking/ibc_module.go (hunk 3 above)
Comment (lehieuhust) [reply x/interchainstaking/ibc_module.go]: Fix in https://github.com/notional-labs/quicksilver/commit/aab37f851670fde663ed1b04b299e6be033aed8d

File: x/interchainstaking/types/codec.go (hunk 5 above)
Comment (lehieuhust) [reply x/interchainstaking/types/codec.go]: Fix in https://github.com/notional-labs/quicksilver/commit/aab37f851670fde663ed1b04b299e6be033aed8d

File: x/interchainstaking/keeper/proposal_handler.go (hunk 4 above)
Comment (lehieuhust) [reply x/interchainstaking/keeper/proposal_handler.go]: Fix in https://github.com/notional-labs/quicksilver/commit/aab37f851670fde663ed1b04b299e6be033aed8d

File: app/app.go (hunk 2 above)
Comment (lehieuhust) [reply app/app.go:618]: This is not require.

**Comment by lehieuhust** [comment]
@nghuyenthevinh2000  Can not find the definition of MsgRedeemTokensforShares, MsgTokenizeSharesResponse, MsgTokenizeShares

**Pull request merged**